### DISCUSSION TITLE
On leveraging Github's "sub-issues" capabilities for tracking dependent work

### DISCUSSION THREAD

**Discussion by julealgon** [dotnet/csharplang#8585] [General] [answered]
The dotnet repos tend to use labels for tracking pending work in a few cases.

One such example is this issue:
- https://github.com/dotnet/csharplang/issues/4304#issuecomment-2474208839

Someone asked a question regarding why the issue wasn't yet closed, and Cyrus responded that it was still not added to ECMA standard, something that is part of the overall work.

That missing "task" is currently being tracked via the [Implemented Needs ECMA Spec](https://github.com/dotnet/csharplang/labels/Implemented%20Needs%20ECMA%20Spec) label, but a label arguably doesn't provide great visibility to work completion.

Has the team considered tapping into the new(ish) ["sub-issues" capabilities introduced in Github](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/adding-sub-issues), to model these "pending tasks" as sub-issues in the main issue?

I feel like that would provide more clarity to readers of the repo on what has been done and what is still pending for completing a given issue.

**Comment by 333fred** [comment, marked as answer]
We do understand that it can be hard to see what the current status of a language feature is, which is why we're currently in the process of revamping the entire csharplang process (https://github.com/dotnet/csharplang/pull/8578 is that). In terms of sub issues, I'm not sure they're a great fit for this, but after the repo rework is done, with champion issues being locked for comment and current status more apparent from the issue, we can take a look at whether sub issues might work.

**Reply by julealgon** [reply]
I see. Looking forward to the improvements that new process will bring. Thanks for the prompt reply @333fred .
